Linking fails for ryzom_gameshare when using WITH_STATIC=OFF

ryzom/common/src/game_share has zlib dependency.

When using WITH_STATIC=OFF, then ryzom_gameshare fails to link

```
/usr/bin/ld: CMakeFiles/ryzom_gameshare.dir/object.cpp.o: in function `R2::CObjectSerializer::uncompressImpl()':
object.cpp:(.text+0x1818a): undefined reference to `uncompress'
/usr/bin/ld: CMakeFiles/ryzom_gameshare.dir/object.cpp.o: in function `R2::CObjectSerializer::compress()':
object.cpp:(.text+0x18c3d): undefined reference to `compress'
```

I think that dependency came from NELMISC_LIBRARIES definition before.

I used this cmake command
```
cmake -B .build -DWITH_RYZOM=ON -DWITH_RYZOM_SERVER=OFF -DWITH_NEL_TOOLS=OFF -DWITH_RYZOM_TOOLS=OFF -DWITH_NEL_SAMPLES=OFF -DWITH_3D=OFF -DWITH_SOUND=OFF -DWITH_RYZOM_CLIENT=OFF -DWITH_PCH=OFF -DWITH_NEL_TESTS=OFF -DWITH_GUI=OFF -DWITH_STATIC=OFF .
```

Obviously `PRIVATE ZLIB::ZLIB` needs to be added to game_share, but maybe CI should also be updated to catch this.

--- a/.devcontainer/Dockerfile
+++ b/.devcontainer/Dockerfile
@@ -1,7 +1,7 @@
 FROM ubuntu:24.04
 RUN touch /var/mail/ubuntu && chown ubuntu /var/mail/ubuntu && userdel --remove ubuntu
 
-RUN apt update; \
+RUN apt update && \
     DEBIAN_FRONTEND=noninteractive \
     apt install --yes \
         autoconf \
@@ -52,7 +52,8 @@ RUN apt install --yes \
         git \
         graphviz \
         libxrandr-dev \
-        wget
+        wget \
+        sudo
 
 
 # work arround as there is no debug version available

--- a/.github/nel/action.yml
+++ b/.github/nel/action.yml
@@ -46,7 +46,7 @@ runs:
     - name: Configure CMake
       if: runner.os == 'Linux'
       shell: bash
-      run: cmake --preset linux-nel-${{ inputs.build-type }}
+      run: cmake --preset ci-linux-nel-${{ inputs.build-type }}
 
     - name: Build
       shell: bash

--- a/.github/nel/action.yml
+++ b/.github/nel/action.yml
@@ -59,7 +59,7 @@ runs:
     - uses: actions/upload-artifact@v4
       if: ( !cancelled() )
       with:
-        name: nel-${{ inputs.os }}-test-results
+        name: nel-${{ inputs.os }}-${{ inputs.build-type }}-test-results
         if-no-files-found: error
         path: ${{ inputs.build-dir }}/reports/*.xml
 

--- a/.github/nel/action.yml
+++ b/.github/nel/action.yml
@@ -1,14 +1,17 @@
 name: build
 description: build
 inputs:
+  os:
+    description: os
+    required: true
+  build-type:
+    description: link statically or dynamically value  should be static | dynamic
+    required: true
   build-dir:
     description: build directory
     required: true
   build-version:
     description: build version
-    required: true
-  os:
-    description: os
     required: true
 runs:
   using: composite
@@ -43,7 +46,7 @@ runs:
     - name: Configure CMake
       if: runner.os == 'Linux'
       shell: bash
-      run: cmake --preset linux-nel
+      run: cmake --preset linux-nel-${{ inputs.build-type }}
 
     - name: Build
       shell: bash

--- a/.github/ryzom/action.yml
+++ b/.github/ryzom/action.yml
@@ -4,6 +4,9 @@ inputs:
   os:
     description: os
     required: true
+  cmake-options:
+    description: cmake options
+    required: false
   install-dir:
     description: install directory
     required: true
@@ -58,7 +61,7 @@ runs:
     - name: Configure CMake
       if: runner.os == 'Linux'
       shell: bash
-      run: cmake --preset ci-linux-ryzom
+      run: cmake --preset ci-linux-ryzom ${{ inputs.cmake-options }}
 
     - name: Build
       shell: bash

--- a/.github/ryzom/action.yml
+++ b/.github/ryzom/action.yml
@@ -4,9 +4,9 @@ inputs:
   os:
     description: os
     required: true
-  cmake-options:
-    description: cmake options
-    required: false
+  build-type:
+    description: link statically or dynamically value  should be static | dynamic
+    required: true
   install-dir:
     description: install directory
     required: true
@@ -61,7 +61,7 @@ runs:
     - name: Configure CMake
       if: runner.os == 'Linux'
       shell: bash
-      run: cmake --preset ci-linux-ryzom ${{ inputs.cmake-options }}
+      run: cmake --preset ci-linux-ryzom-${{ inputs.build-type }}
 
     - name: Build
       shell: bash

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -53,6 +53,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-22.04, ubuntu-24.04 ]
+        build-type: ["static", "dynamic"]
     steps:
       - uses: actions/checkout@v4
 
@@ -65,13 +66,14 @@ jobs:
       - uses: ./.github/nel
         if: steps.cache.outputs.cache-hit != 'true'
         with:
+          os: ${{ matrix.os }}
+          build-type: ${{ matrix.build-type }}
           build-dir: ${{ needs.strings.outputs.build-dir }}
           build-version: ${{ needs.strings.outputs.build-version }}
-          os: ${{ matrix.os }}
 
       - uses: actions/upload-artifact@v4
         with:
-          name: nel-${{ matrix.os }}-${{ needs.strings.outputs.build-version }}
+          name: nel-${{ matrix.os }}-${{ matrix.build-type }}-${{ needs.strings.outputs.build-version }}
           path: ${{ needs.strings.outputs.build-dir }}/nel/nel-*
           if-no-files-found: error
 
@@ -89,7 +91,7 @@ jobs:
 
       - uses: actions/download-artifact@v4
         with:
-          pattern: nel-${{ matrix.os }}-*
+          pattern: nel-${{ matrix.os }}-static-*
           merge-multiple: true
           path: ${{ needs.strings.outputs.build-dir }}
 
@@ -129,7 +131,7 @@ jobs:
 
       - uses: actions/download-artifact@v4
         with:
-          pattern: nel-${{ matrix.os }}-*
+          pattern: nel-${{ matrix.os }}-${{ matrix.build-type }}-*
           merge-multiple: true
           path: ${{ needs.strings.outputs.build-dir }}
 
@@ -151,9 +153,8 @@ jobs:
           build-version: ${{ needs.strings.outputs.build-version }}
 
       - uses: actions/upload-artifact@v4
-        if: matrix.build-type == 'static'
         with:
-          name: ryzom-${{ matrix.os }}-${{ needs.strings.outputs.build-version }}
+          name: ryzom-${{ matrix.os }}-${{ matrix.build-type }}-${{ needs.strings.outputs.build-version }}
           path: ${{ needs.strings.outputs.build-dir }}/ryzom/ryzom-*
           if-no-files-found: error
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -63,7 +63,7 @@ jobs:
       - id: cache
         uses: actions/cache@v4
         with:
-          key: nel-${{ matrix.os }}-${{ hashFiles('nel/**', '.github/nel/**', 'cmake/**', 'CMakeModules/**', 'CMakeLists.txt', 'CMakePresets.json') }}
+          key: nel-${{ matrix.os }}-${{ matrix.build-type }}-${{ hashFiles('nel/**', '.github/nel/**', 'cmake/**', 'CMakeModules/**', 'CMakeLists.txt', 'CMakePresets.json') }}
           path: ${{ needs.strings.outputs.build-dir }}/nel/nel-*
 
       - uses: ./.github/nel

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -174,7 +174,7 @@ jobs:
 
       - uses: actions/download-artifact@v4
         with:
-          pattern: nel-${{ matrix.os }}-*
+          pattern: nel-${{ matrix.os }}-static-*
           merge-multiple: true
           path: ${{ needs.strings.outputs.build-dir }}
 
@@ -212,7 +212,7 @@ jobs:
 
       - uses: actions/download-artifact@v4
         with:
-          pattern: nel-${{ matrix.os }}-*
+          pattern: nel-${{ matrix.os }}-static-*
           merge-multiple: true
           path: ${{ needs.strings.outputs.build-dir }}
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -126,6 +126,7 @@ jobs:
     needs: [ strings, nel ]
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
         os: [ ubuntu-24.04 ]
         build-type: ["static", "dynamic"]

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -2,7 +2,7 @@ name: "Ryzom Core: CI / CD"
 
 on:
   push:
-    branches: [ "core4", "feature/*" ]
+    branches: [ "core4", "feature/*", "bugfix/*" ]
     paths-ignore:
       - '.devcontainer/**'
       - '.github/workflows/**'

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -123,6 +123,7 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-24.04 ]
+        cmake-options: ["-DWITH_STATIC=ON", "-DWITH_STATIC=OFF"]
     steps:
       - uses: actions/checkout@v4
 
@@ -142,6 +143,7 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true'
         with:
           os: ${{ matrix.os }}
+          cmake-options: ${{ matrix.cmake-options }}
           install-dir: ${{ needs.strings.outputs.install-dir }}
           image-group: ${{ needs.strings.outputs.image-group }}
           image-version: ${{ needs.strings.outputs.image-version }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -54,6 +54,9 @@ jobs:
       matrix:
         os: [ ubuntu-22.04, ubuntu-24.04 ]
         build-type: ["static", "dynamic"]
+        exclude:
+          - os: ubuntu-22.04
+            build-type: dynamic
     steps:
       - uses: actions/checkout@v4
 

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -123,7 +123,7 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-24.04 ]
-        cmake-options: ["-DWITH_STATIC=ON", "-DWITH_STATIC=OFF"]
+        build-type: ["static", "dynamic"]
     steps:
       - uses: actions/checkout@v4
 
@@ -143,7 +143,7 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true'
         with:
           os: ${{ matrix.os }}
-          cmake-options: ${{ matrix.cmake-options }}
+          build-type: ${{ matrix.build-type }}
           install-dir: ${{ needs.strings.outputs.install-dir }}
           image-group: ${{ needs.strings.outputs.image-group }}
           image-version: ${{ needs.strings.outputs.image-version }}
@@ -151,6 +151,7 @@ jobs:
           build-version: ${{ needs.strings.outputs.build-version }}
 
       - uses: actions/upload-artifact@v4
+        if: matrix.build-type == 'static'
         with:
           name: ryzom-${{ matrix.os }}-${{ needs.strings.outputs.build-version }}
           path: ${{ needs.strings.outputs.build-dir }}/ryzom/ryzom-*

--- a/nel/CMakePresets.json
+++ b/nel/CMakePresets.json
@@ -40,7 +40,19 @@
       "inherits": [
         "linux-nel"
       ],
-      "name": "ci-linux-nel"
+      "name": "ci-linux-nel-static",
+      "cacheVariables": {
+        "WITH_STATIC": true
+      }
+    },
+    {
+      "inherits": [
+        "linux-nel"
+      ],
+      "name": "ci-linux-nel-dynamic",
+      "cacheVariables": {
+        "WITH_STATIC": false
+      }
     },
     {
       "inherits": [
@@ -59,7 +71,7 @@
     {
       "inherits": "nel",
       "name": "ci-linux-nel",
-      "configurePreset": "ci-linux-nel"
+      "configurePreset": "ci-linux-nel-dynamic"
     },
     {
       "inherits": "ci-linux-nel",
@@ -88,7 +100,7 @@
       "steps": [
         {
           "type": "configure",
-          "name": "ci-linux-nel"
+          "name": "ci-linux-nel-dynamic"
         },
         {
           "type": "build",

--- a/nel/CMakePresets.json
+++ b/nel/CMakePresets.json
@@ -38,6 +38,12 @@
     },
     {
       "inherits": [
+        "linux-nel"
+      ],
+      "name": "ci-linux-nel"
+    },
+    {
+      "inherits": [
         "windows",
         "nel"
       ],
@@ -49,6 +55,18 @@
       "inherits": "release",
       "name": "nel",
       "configurePreset": "nel"
+    },
+    {
+      "inherits": "nel",
+      "name": "ci-linux-nel",
+      "configurePreset": "ci-linux-nel"
+    },
+    {
+      "inherits": "ci-linux-nel",
+      "name": "ci-linux-nel/install",
+      "targets": [
+        "nel/install"
+      ]
     }
   ],
   "testPresets": [
@@ -63,5 +81,24 @@
       "name": "nel",
       "configurePreset": "nel"
     }
+  ],
+  "workflowPresets": [
+    {
+      "name": "ci-linux-nel",
+      "steps": [
+        {
+          "type": "configure",
+          "name": "ci-linux-nel"
+        },
+        {
+          "type": "build",
+          "name": "ci-linux-nel"
+        },
+        {
+          "type": "build",
+          "name": "ci-linux-nel/install"
+        }
+      ]
+   }
   ]
 }

--- a/ryzom/CMakePresets.json
+++ b/ryzom/CMakePresets.json
@@ -46,7 +46,21 @@
         "ci-ryzom",
         "ryzom"
       ],
-      "name": "ci-linux-ryzom"
+      "name": "ci-linux-ryzom-static",
+      "cacheVariables": {
+        "WITH_STATIC": true
+      }
+    },
+    {
+      "inherits": [
+        "linux",
+        "ci-ryzom",
+        "ryzom"
+      ],
+      "name": "ci-linux-ryzom-dynamic",
+      "cacheVariables": {
+        "WITH_STATIC": false
+      }
     },
     {
       "inherits": [

--- a/ryzom/CMakePresets.json
+++ b/ryzom/CMakePresets.json
@@ -75,6 +75,18 @@
       "inherits": "release",
       "name": "ryzom",
       "configurePreset": "ryzom"
+    },
+    {
+      "inherits": "ryzom",
+      "name": "ci-linux-ryzom",
+      "configurePreset": "ci-linux-ryzom-dynamic"
+    },
+    {
+      "inherits": "ci-linux-ryzom",
+      "name": "ci-linux-ryzom/install",
+      "targets": [
+        "ryzom/install"
+      ]
     }
   ],
   "testPresets": [
@@ -88,6 +100,25 @@
     {
       "name": "ryzom",
       "configurePreset": "ryzom"
+    }
+  ],
+  "workflowPresets": [
+    {
+      "name": "ci-linux-ryzom",
+      "steps": [
+        {
+          "type": "configure",
+          "name": "ci-linux-ryzom-dynamic"
+        },
+        {
+          "type": "build",
+          "name": "ci-linux-ryzom"
+        },
+        {
+          "type": "build",
+          "name": "ci-linux-ryzom/install"
+        }
+      ]
     }
   ]
 }

--- a/ryzom/client/src/CMakeLists.txt
+++ b/ryzom/client/src/CMakeLists.txt
@@ -204,7 +204,6 @@ IF(WITH_RYZOM_CLIENT)
         ${LUA_INCLUDE_DIR}
         ${LUABIND_INCLUDE_DIR}
         ${CURL_INCLUDE_DIRS}
-        ${ZLIB_INCLUDE_DIR}
         ${OPENSSL_INCLUDE_DIR}
   )
 
@@ -232,6 +231,7 @@ IF(WITH_RYZOM_CLIENT)
               ${LUABIND_LIBRARIES}
               CURL::libcurl
               ${OPENSSL_LIBRARIES}
+              ZLIB::ZLIB
   )
 
   IF(NOT APPLE AND NOT WIN32)

--- a/ryzom/common/src/game_share/CMakeLists.txt
+++ b/ryzom/common/src/game_share/CMakeLists.txt
@@ -44,6 +44,8 @@ TARGET_LINK_LIBRARIES(ryzom_gameshare
 		NeL::net
 		NeL::ligo
 		NeL::georges
+		#PRIVATE
+		#ZLIB::ZLIB
 )
 
 NL_DEFAULT_PROPS(ryzom_gameshare "Ryzom, Library: Game Share")

--- a/ryzom/common/src/game_share/CMakeLists.txt
+++ b/ryzom/common/src/game_share/CMakeLists.txt
@@ -36,7 +36,7 @@ LIST(REMOVE_ITEM HEADER ${CMAKE_CURRENT_SOURCE_DIR}/enum_template.h)
 
 NL_TARGET_LIB(ryzom_gameshare ${HEADER} ${PRIV_TIME_H} ${SRC} ${SRC_TIME} ${R2})
 
-TARGET_INCLUDE_DIRECTORIES(ryzom_gameshare PRIVATE ${ZLIB_INCLUDE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
+TARGET_INCLUDE_DIRECTORIES(ryzom_gameshare PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 
 TARGET_LINK_LIBRARIES(ryzom_gameshare
 		PUBLIC
@@ -44,8 +44,8 @@ TARGET_LINK_LIBRARIES(ryzom_gameshare
 		NeL::net
 		NeL::ligo
 		NeL::georges
-		#PRIVATE
-		#ZLIB::ZLIB
+		PRIVATE
+		ZLIB::ZLIB
 )
 
 NL_DEFAULT_PROPS(ryzom_gameshare "Ryzom, Library: Game Share")

--- a/ryzom/tools/client/client_patcher/CMakeLists.txt
+++ b/ryzom/tools/client/client_patcher/CMakeLists.txt
@@ -12,7 +12,6 @@ ADD_EXECUTABLE(ryzom_client_patcher ${SRC})
 TARGET_INCLUDE_DIRECTORIES(ryzom_client_patcher
         PRIVATE
         ${CURL_INCLUDE_DIRS}
-        ${ZLIB_INCLUDE_DIR}
         ${CMAKE_SOURCE_DIR}/ryzom/client/src
 )
 
@@ -24,6 +23,7 @@ TARGET_LINK_LIBRARIES(ryzom_client_patcher
         ryzom_gameshare
         NeL::sevenzip
         CURL::libcurl
+        ZLIB::ZLIB
 )
 
 IF (APPLE)

